### DISCUSSION TITLE
How can to prevent the creation of code block by using tab key?

### DISCUSSION THREAD

**Discussion by jongjunpark** [remarkjs/remark#623] [Q&A] [answered]
I don't want to create code block by using tab key.
below text creates a code block in react-markdown.
```
    text
```

I want It don't create a code block and I just want to apply tab size in react-markdown.
How can I prevent and custom this?

**Comment by ChristianMurphy** [comment, marked as answer]
https://stackoverflow.com/questions/6046263/how-to-indent-a-few-lines-in-markdown-markup#6046472

**Reply by jongjunpark** [reply]
How Can I always apply "\&nbsp;" when I use a tab key?

**Reply by ChristianMurphy** [reply]
you could remap your keyboard, or you could not use tabs.

At a core level, indented code blocks are a part of markdown https://spec.commonmark.org/0.29/#indented-code-blocks
You could try the approaches outlined in https://github.com/micromark/micromark/discussions/31 and https://github.com/remarkjs/remark/issues/552 to disable a core construct, but it may be a bit complex.

**Reply by jongjunpark** [reply]
Thanks!

**Reply by wooorm** [reply]
Do you use template strings around your markdown? (With the ticks)? Is that why you want to not support tabs?
